File: src/Gateway/Domovoy.DbGateway/Models/Sensor.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Domovoy.DbGateway.Models;

public class Sensor
{
    [BsonId]
    public string SensorId { get; set; } = null!;

    public string DeviceId { get; set; } = null!;
    
    public string Type { get; set; } = null!;
    
    public int UpdateFrequency { get; set; } = 60; // seconds
    
    public double Precision { get; set; } = 0.1;
    
    public double? LastValue { get; set; }
}

File: src/UI/WebUI/src/types/sensor.ts
// Sensor Types for WebUI
// Validates: Requirements 3.1, 4.1

export type SensorType = 'Temperature' | 'Humidity' | 'Motion' | 'Light' | 'Pressure';

export interface Sensor {
  sensorId: string;
  deviceId: string;
  type: SensorType;
  updateFrequency: number;
  precision: number;
  lastValue: number | null;
}

export interface SensorReading {
  sensorId: string;
  timestamp: Date;
  value: number;
  unit: string;
}
